src/olympia/api/fields.py

```diff
@@ -102,7 +102,10 @@ def fetch_single_translation(self, obj, source, field, requested_language):
 
     def get_attribute(self, obj):
         source = self.source or self.field_name
-        field = fields.get_attribute(obj, source.split('.'))
+        try:
+            field = fields.get_attribute(obj, source.split('.'))
+        except AttributeError:
+            field = None
```

Comment:
I can't really grasp in which cases that happen, mind to explain a little? Ideally adding a test to `api/tests/test_fields.py` to make sure we don't just test this change via blocklist tests since it's a more generic API that can be used by other parts too.

Reply:
if your source has nested attributes (e.g. `source='addon.current_version.release_notes'`) but one of the parent objects is `None` (e.g. `addon.current_version is None`) there's an exception thrown.   

We need it for this patch because there's an edge case where a Block exists but the Addon guid doesn't so a source of `addon.name` can raise - I guess it's never been an issue before because we generally include entire serializers for child objects which safely handle it.

Reply:
https://github.com/mozilla/addons-server/pull/13786/commits/e104df29ae5b230b3db3e3af9f4c052d5d04f40c